sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -254,3 +254,54 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+object Logarithm {
+  def apply(child: Expression): Expression = new Log(child)
+}
+
+case class Logarithm(left: Expression, right: Expression)
```

Comment:
We seem to be doing this throughout the file, but it seems pretty confusing to me to be using `left` and `right` instead of something like `value` and `base`.  I'm not sure this is worth whatever code reuse we are getting.